azure-iot-device/tests/iothub/test_inbox_manager.py

```diff
@@ -12,52 +12,86 @@
 from azure.iot.device.iothub.models import Message, MethodRequest
 
 inbox_type_list = []
+inbox_type_ids = []
 
 # Create a list of Inbox types to test with the manager
 try:
     from azure.iot.device.iothub.aio.async_inbox import AsyncClientInbox
 
     inbox_type_list.append(AsyncClientInbox)
+    inbox_type_ids.append("Configured with AsyncClientInboxes")
 except SyntaxError:
     # AsyncClientInbox not available if Python < 3.5
     pass
 finally:
     from azure.iot.device.iothub.sync_inbox import SyncClientInbox
 
     inbox_type_list.append(SyncClientInbox)
+    inbox_type_ids.append("Configured with SyncClientInboxes")
 
 
+@pytest.fixture(params=inbox_type_list, ids=inbox_type_ids)
+def inbox_type(request):
+    return request.param
+
+
+@pytest.fixture
+def manager(inbox_type):
+    return InboxManager(inbox_type=inbox_type)
+
+
+@pytest.fixture
+def message():
+    return Message("some message")
+
+
+@pytest.fixture
+def method_request():
+    return MethodRequest(request_id="1", name="some_method", payload="{'key': 'value'}")
+
+
+@pytest.mark.describe("InboxManager")
 class TestInboxManager(object):
-    @pytest.fixture(params=inbox_type_list)
-    def inbox_type(self, request):
-        return request.param
+    @pytest.mark.it("Instantiates with an empty C2D inbox")
+    def test_instantiates_with_empty_c2d_inbox(self, manager):
```

Comment:
>manager [](start = 53, length = 7)

I know that you know this, but I'm pointing out that you're using a parameterized fixture for the object that you're testing and that's a no-no.  Too bad there's not a syntax to use a parameterized fixture (or any fixture) as a class modifier

Reply:
OK, we talked about this one yesterday.  This is the weird one because the parameter is the type of inbox, not the manager itself.  

---
In reply to: [290984609](https://github.com/Azure/azure-iot-sdk-python-preview/pull/101#discussion_r290984609) [](ancestors = 290984609)

Reply:
Yeah, it's parametrizing different configurations of the same unit under test. The antipattern to avoid is using parametrization to cover multiple units, but since this one is the same unit, it's fine (and in fact, desirable)

---
In reply to: [291182134](https://github.com/Azure/azure-iot-sdk-python-preview/pull/101#discussion_r291182134) [](ancestors = 291182134,290984609)